client/layout/sidebar/item.jsx

```diff
@@ -68,7 +68,7 @@ export default React.createClass( {
 					<span className="menu-link-text">{ this.props.label }</span>
 					{ isExternalLink ? <span className="noticon noticon-external" /> : null }
 				</a>
-				{ this.renderButton( this.props.buttonLink ) }
+				{ this.renderButton( this.props.buttonLink ) || this.props.children }
```

Comment:
Should the `<a>` in `renderButton` be updated to apply `sidebar__button` class instead of `add-new` (especially since we removed the styles for the latter).